tls/s2n_ktls.c

```diff
@@ -105,8 +192,7 @@ static S2N_RESULT s2n_ktls_configure_socket(struct s2n_connection *conn, s2n_ktl
      * This is a safe and non destructive operation on Linux.
      */
     int ret = setsockopt(fd, S2N_SOL_TCP, S2N_TCP_ULP, S2N_TLS_ULP_NAME, S2N_TLS_ULP_NAME_SIZE);
-
-    if (ret != 0) {
+    if (ret < 0) {
         /* https://man7.org/linux/man-pages/man3/errno.3.html
          * EEXIST indicates that TCP_ULP has already been enabled on the socket.
          * This is a noop and therefore safe to ignore.
```

Comment:
You should be linking to the man page for setsockopt, not errno. Individual methods are the final authority on their own behavior. For example, the [setsockopt man page](https://man7.org/linux/man-pages/man3/errno.3.html) does say that ret != 0 should be the right check, but it also doesn't appear to set EEXIST (unless that's added when ktls support is added?)

Reply:
This was observed behavior and I would need to find ktls specific docs for this :`)

Reply:
Here is the source and a test which confirm this behavior. I dont think documentation exits though.. its pretty sparse for kTLS

- https://github.com/torvalds/linux/blob/831fe284d8275987596b7d640518dddba5735f61/net/ipv4/tcp_ulp.c#L64-L65
- https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/net/tls.c#L1695